### PR TITLE
CAL: Allow for multiple slugs

### PR DESCRIPTION
**Pull request by waptik**
Prior to this PR, the inspector was pulling event data from cal.com's api for the default event slug that everyone gets upon signup `15min`. If you do not have any event with the `15min` slug then no data will be returned when you use your username.
So this PR aims to allow `Users` set their own event slug alongside their username.

- Set default username to `cal.com` and eventSlug to `15min`
- Fetch initial data when `config.name` isn't available, meaning it's freshly created frame without any data that was fetch from the api

https://www.loom.com/share/e6f2fe5108e849c292ea3b45d0e46640

### CODE REVIEW

**Comment by FTCHD** [comment]
The current version lets the Viewer select between `15min` and `30min` options of the User's cal frame, so allowing only one option is a downgrade.

You could let the User input multiple slugs (up to 4) when configuring the frame, and in case only 1 slug is present in the config then we skip the timeframe selection screen entirely in the handler/function, if not we show the usual screen and let the Viewer select.

**Comment by waptik** [comment]
> The current version lets the Viewer select between `15min` and `30min` options of the User's cal frame, so allowing only one option is a downgrade.
> 
> You could let the User input multiple slugs (up to 4) when configuring the frame, and in case only 1 slug is present in the config then we skip the timeframe selection screen entirely in the handler/function, if not we show the usual screen and let the Viewer select.

Sorry i changed the title without noticing you did it as i thought i was a mistake from my part when creating the PR.
I haven't deeply tried this template. When i was working on #65, i couldn't get the preview frame to fetch my information because i didn't have an event with `15min` slug so figured it was technical glitch hence why i opted to do these changes.

I'll work on the multiple slugs selection as suggested, today.

**Comment by FTCHD** [comment]
Got it, seems like @LeoFranklin015 submitted an incomplete template then, or at least untested.

**Comment by waptik** [comment]
> Got it, seems like @LeoFranklin015 submitted an incomplete template then, or at least untested.

And a lot more code repetition. Did you go through the entire template's codebase?

**Comment by FTCHD** [comment]
I did and also ran it myself, but shamefully not at the depth necessary it seems, and that's on me.

My primary focus is finding external libraries/external endpoints or anything that would be a security problem like obfuscated/weird code, and trusted the functionality part on the contributor and their demo. IMO, bugs are normal in SWE and it's why `Issues` exist, but this looks like a crass oversight more than a smol bug.

It's also not a rocket science type of template (like @ryanjshaw's Figma template for example, which allows you to use Figma files to create Frames), so I didn't assume such spaghetti code and errors were there as they did not happen for me, but I have both `15min` and `30min` meeting types in my Cal account so that's why.

Funnily enough the Figma template, which is the most complex one we have currently, is also the most cleanly written and organized, so there's a lesson here somewhere.

Anyways, let's do it properly this time and let the User add their meeting types (slugs).

**Comment by waptik** [comment]
Here's an updated demo video
https://www.loom.com/share/6d6b078680684b9ba0e9b255f23b06aa

**Comment by FTCHD** [comment]
Looks good, is there a way to check if the `slug` exists before adding it? 

I see that we can simply query `https://cal.com/USER/SLUG` and if it returns `404` it means the slug does not exist. When "ADD" is pressed, we check first, and only add it if it exists, otherwise we fire a `toast.error()` in the Inspector.

**Comment by waptik** [comment]
> Looks good, is there a way to check if the `slug` exists before adding it?
Thanks. There's a handler for it

> I see that we can simply query `https://cal.com/USER/SLUG` and if it returns `404` it means the slug does not exist. When "ADD" is pressed, we check first, and only add it if it exists, otherwise we fire a `toast.error()` in the Inspector.

So when you add a slug, [this block](https://github.com/FTCHD/frametrain/blob/a4aeb2bdf90bba398ad2cd921e3fa8bcf500dc9c/templates/cal/Inspector.tsx#L145-L158) is where it's being handled by querying the event's public api endpoint instead of the event page itself and doing all the necessary checks. The slug will only be added if `json` is not null. Took me a while to figure it whilst playing around with the api.

I suppose i need to fire the `toast.error` to draw the `Users`'s attention. I figured as a user, you ought to know if you have an event with such a slug or not; hence why in the video i switch to an event's tab to copy the username(i previously tested with all the slugs available)

File: templates/cal/Inspector.tsx [hunk 1]
```diff
@@ -142,37 +143,43 @@ export default function Inspector() {
...
+                                                slug: slug,
+                                                duration: duration,
+                                                formattedDuration: getDurationFormatted(duration),
+                                            },
+                                        ]
 
-                                    setLoading(false)
+                                        updateConfig({ events: newEvents })
+                                    } catch (_) {
+                                        toast.error(`No Event Type found for: ${eventSlug}`)
```
Comment (FTCHD) [inline templates/cal/Inspector.tsx:177]: Please simply write it as "No event type found for....", no reason to capitalize those words. 

Everything else looks fine!

**Comment by waptik** [comment]
Noticed some hosts have a really long bio such as https://cal.com/ajayanbisi
Working on a fix now
image

**Comment by waptik** [comment]
> Noticed some hosts have a really long bio such as https://cal.com/ajayanbisi Working on a fix now image


UPDATE:

It has been fixed to only show the first 90 characters of each child element
image

**Comment by waptik** [comment]
Closing this in favor of #72

**Pull request closed without merging**